src/platforms/zephyr/src/lib/sys.c

```diff
@@ -0,0 +1,322 @@
+/*
+ * This file is part of AtomVM.
+ *
+ * Copyright 2023 Winford (Uncle Grumpy]) <winford@object.stream>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later
+ */
+
+#include <zephyr/posix/time.h>
+
+#include <avmpack.h>
+#include <defaultatoms.h>
+#include <scheduler.h>
+#include <sys.h>
+
+// #define ENABLE_TRACE
+#include <trace.h>
+
+#include <zephyr/kernel.h>
+
+// #include <libopencm3/cm3/nvic.h>
```

Comment:
Should commented code of the libopencm3-based version remain?

Reply:
Not at all, I left some of that for reference while it's in draft, just so I could try to make sure this port starts on parity with the libopencm3 stm32 port. I have a lot of commented out code to remove still.